XenCert/StorageHandler.py

```diff
@@ -3170,3 +3169,100 @@ def ControlPathStressTests(self):
         XenCertPrint("Checkpoints: %d, totalCheckPoints: %s" % (checkPoint, totalCheckPoints))
 
         return (retVal, checkPoint, totalCheckPoints)
+
+    def DataIntegrityTests(self):
+        retVal = None
+        checkPoint = 0
+        totalCheckPoints = 11
+
+        vm_uuid = StorageHandlerUtil._get_localhost_uuid()
+        XenCertPrint("Got vm_uuid as %s" % vm_uuid)
+        vm_ref = self.session.xenapi.VM.get_by_uuid(vm_uuid)
+        sr_ref = None
+        vdi_ref = None
+        vbd_ref = None
+
+        try:
+            #1) Create SR
+            (retVal, sr_ref, dconf) = self.Create()
+            Print("Created SR")
+            checkPoint += 1
+
+            #2) Create 4GB VDI in SR
+            (retVal, vdi_ref) = self.Create_VDI(sr_ref,
```

Comment:
Create_VDI doesn't raise an exception and we aren't checking the retVal. The results are interesting if VDI create fails for some reason. We increment the checkpoint (now 2), fail in VBD attach and in cleanup try to destroy the VDI (because checkpoint > 1) that doesn't exist. The whole cleanup process fails and we are left behind with the SR. We should handle the error with this call.